src/components/RightPane/CoursePane/SearchForm/DeptSearchBar/DeptSearchBar.tsx

```diff
@@ -48,8 +60,17 @@ class DeptSearchBar extends PureComponent<DeptSearchBarProps, DeptSearchBarState
         }
         this.state = {
             value: {
-                deptValue: RightPaneStore.getFormData().deptValue,
-                deptLabel: RightPaneStore.getFormData().deptLabel,
+                deptValue: RightPaneStore.getUrlDeptValue() != "null" && RightPaneStore.getUrlDeptValue() != "" && RightPaneStore.getUrlDeptValue() != " " 
+                ?
+                this.updatedeptValueAndGetFormData()
+                : 
+                RightPaneStore.getFormData().deptValue,
+
+                deptLabel: RightPaneStore.getUrlDeptLabel() != "null" && RightPaneStore.getUrlDeptLabel() != "" && RightPaneStore.getUrlDeptLabel() != " " 
+                ?
+                this.updatedeptLabelAndGetFormData()
+                : 
+                RightPaneStore.getFormData().deptLabel,
```

Comment:
Can you do the same thing that I recommended for the CourseNumbeSearch? e.g. create a method that will abstract away all of these steps. Another benefit of this is that you're re-calculating function calls multiple times; having a dedicated function allows you to save them into local variables